tests/unit/specs/components/luxLibraryHeader.spec.js

```diff
@@ -0,0 +1,46 @@
+import { mount } from "@vue/test-utils"
+import LuxLibraryHeader from "@/components/LuxLibraryHeader.vue"
+
+describe("LuxLibraryHeader.vue", () => {
+  let wrapper
+  beforeEach(() => {
+    wrapper = mount(LuxLibraryHeader, {
+      attachTo: document.body,
+      props: {
+        type: "abbr",
+        appName: "My Application",
+        abbrName: "My App",
+        appUrl: "http://example.com/",
+        maxWidth: 1111,
+        theme: "light",
+      },
+    })
+  })
+  it("accepts a type prop", () => {
+    expect(wrapper.find("abbr").text()).toBe(
+      "Princeton University Library LogoPrinceton University Library IconMy ApplicationMy App"
+    )
+  })
+  it("gets the theme and defaults to dark unless the theme is light", () => {
+    expect(wrapper.vm.value(wrapper.vm.theme)).toEqual("light")
+    expect(wrapper.vm.value("light")).toEqual("light")
+    expect(wrapper.vm.value("dark")).toEqual("dark")
+    expect(wrapper.vm.value("shade")).toEqual("dark")
+  })
+  it("has a long and short name", () => {
+    let full_name = wrapper.find(".full-name")
+    expect(full_name.text()).toEqual("My Application")
+    // expect(full_name.isVisible()).toBe(false)
```

Comment:
Do we want to keep these commented-out lines?

Reply:
I think it would be nice to test that it's showing the correct name given the screen size, but I was struggling with figuring out how to configure it to compile the CSS, and I wasn't confident that we wanted to do that for all the tests.

I was thinking about leaving the commented out tests for now and having the CSS stuff be future work? But maybe a separate pending test would be better and clearer? What do you think?

Reply:
I think it would be better to create a new ticket and add the link as a TODO in the spec.

Reply:
Done.